### PR TITLE
Stateful application+ansible

### PR DESCRIPTION
**Pull request by acambera-nearsoft**

### CODE REVIEW

**Comment by acambera-nearsoft** [comment]
This PR includes Axel's changes.

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ bucket_name=terraform-states.nearsoft
 # Choose in case you have multiple profiles on your workstation
 aws_profile ?= nstask
 # Switch to different TF image version
-tf_versions ?= 0.12.21
+tf_versions ?= 0.12.29
```
Comment (joshuansu0897) [inline Makefile:6]: Why we needed to change the version of terraform?

File: modules/ami/variables.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+variable "ami_name" {
+  description = "Name of the AMI to use"
+  type = string
```
Comment (joshuansu0897) [inline modules/ami/variables.tf:3]: i think this is not necessary in the new version (the current version we are working)

this works, is just a comment

File: platform/stateful-application/ansible/roles/mysql/.travis.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  # Create ansible.cfg with correct roles_path
+  - printf '[defaults]\nroles_path=../' >ansible.cfg
+
+script:
+  # Basic role syntax check
+  - ansible-playbook tests/test.yml -i tests/inventory --syntax-check
+
+notifications:
+  webhooks: https://galaxy.ansible.com/api/v1/notifications/
```
Comment (joshuansu0897) [inline platform/stateful-application/ansible/roles/mysql/.travis.yml]: Maybe is a stupid question, but, are we using TravisCI?

I mean, I know this config file is for TravisCI, but, what is the reason to use travis? (i don't have problem with travis, we can use travis, but i want to know the reason)

File: platform/stateful-application/ansible/roles/wordpress/.travis.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  # Create ansible.cfg with correct roles_path
+  - printf '[defaults]\nroles_path=../' >ansible.cfg
+
+script:
+  # Basic role syntax check
+  - ansible-playbook tests/test.yml -i tests/inventory --syntax-check
+
+notifications:
+  webhooks: https://galaxy.ansible.com/api/v1/notifications/
```
Comment (joshuansu0897) [inline platform/stateful-application/ansible/roles/wordpress/.travis.yml]: Maybe is a stupid question, but, are we using TravisCI?

I mean, I know this config file is for TravisCI, but, what is the reason to use travis? (i don't have problem with travis, we can use travis, but i want to know the reason)

**duplicated comment**

File: platform/stateful-application/backend.tf [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    region = "us-east-1"
+    dynamodb_table = "terraform_states"
+  }
+}
+
+data "terraform_remote_state" "cloud" {
+  backend = "s3"
+  config = {
+#    bucket  = "terraform-states.nearsoft"
+    bucket  = "alfredo-terraform-states.nearsoft"
```
Comment (joshuansu0897) [inline platform/stateful-application/backend.tf]: i think we need to use the same statement, i think you change this for your own test, but, i the real project this should be the same state like the others or user envs variables

we can talk about this

File: Makefile (hunk 1 above)
Comment (acambera-nearsoft) [reply Makefile:6]: Terraform wouldn't assign a public IP address to the instance behind the ASG. I updated Terraform and it worked.

File: modules/ami/variables.tf (hunk 2 above)
Comment (acambera-nearsoft) [reply modules/ami/variables.tf:3]: I added to so we could use any other OS version. Current was Ubuntu 18.04, I'm using 20.04 but I didn't want to hardcode into the module.

File: platform/stateful-application/ansible/roles/mysql/.travis.yml (hunk 3 above)
Comment (acambera-nearsoft) [reply platform/stateful-application/ansible/roles/mysql/.travis.yml]: I created the ansible roles using ansible-galexy command line tools and It creates the scaffold of the module, including the travisCI configuration file. Just deleted both TravisCI configuration files

File: platform/stateful-application/ansible/roles/wordpress/.travis.yml (hunk 4 above)
Comment (acambera-nearsoft) [reply platform/stateful-application/ansible/roles/wordpress/.travis.yml]: I created the ansible roles using ansible-galexy command line tools and It creates the scaffold of the module, including the travisCI configuration file. Just deleted both TravisCI configuration files

File: platform/stateful-application/backend.tf (hunk 5 above)
Comment (acambera-nearsoft) [reply platform/stateful-application/backend.tf]: You are right. This is a typo. I was using a different bucket for testing. Just updated the bucket name to the correct one.

**Pull request merged**